src/FuncSharp/Numeric/NumberExtensions.cs

```diff
@@ -61,4 +61,37 @@ public static NonPositiveDecimal AsUnsafeNonPositive(this decimal value)
     {
         return NonPositiveDecimal.CreateUnsafe(value);
     }
+
+    public static decimal SafeDivide(this int a, decimal b, decimal otherwise = 0)
+    {
+        return Divide(a, b).GetOrElse(otherwise);
+    }
+
+    public static decimal SafeDivide(this decimal a, decimal b, decimal otherwise = 0)
+    {
+        return Divide(a, b).GetOrElse(otherwise);
+    }
+
+    public static IOption<decimal> Divide(this int a, decimal b)
+    {
+        return Divide((decimal)a, b);
+    }
+
+    public static IOption<decimal> Divide(this decimal a, decimal b)
```

Comment:
in some cases you return `IOption` and in some cases `?` nullable type, it's inconsistent. Also, why do have to stick with `IOption` and not `Option` itself?

Reply:
Everything returns IOption so that covariance works. It's actually pretty handy. 

It will all change when the fire nation attacks and makes it a struct :)